### PR TITLE
Remove sun.java dependency

### PR DESCRIPTION
**Pull request by hwupathum**
sun.java implementation has been changed with JDK21. Therefore need to move to bouncycastle implementation

### CODE REVIEW

**Comment by hwupathum** [comment]
This is compatible with bc-fips provider as mentioned in https://downloads.bouncycastle.org/fips-java/docs/BCFipsIn100.pdf

**Comment by indeewari** [comment]
Did we check with both FIPS mode and default ?

**Comment by hwupathum** [comment]
> Did we check with both FIPS mode and default ?

Yes

File: pom.xml [hunk 1]
```diff
@@ -997,7 +997,7 @@
 
         <bcprov-jdk18.version>1.74.0.wso2v1</bcprov-jdk18.version>
         <bcpkix-jdk18.version>1.74.0.wso2v1</bcpkix-jdk18.version>
-        <org.bouncycastle.imp.pkg.version.range>[1.52.0,2.0.0)</org.bouncycastle.imp.pkg.version.range>
+        <org.bouncycastle.imp.pkg.version.range>[0.0.0,2.0.0)</org.bouncycastle.imp.pkg.version.range>
```
Comment (indeewari) [inline pom.xml:1000]: All 1.x bc versions are vulnerable.
https://mvnrepository.com/artifact/org.bouncycastle/bc-fips/1.0.2.4

What is the effort of 2.0.0 ?

Comment (hwupathum) [reply pom.xml:1000]: Will track that in a separate effort https://github.com/wso2/product-is/issues/21151

**Review by indeewari — APPROVED** [review]

**Pull request merged**